computedfields/resolver.py

```diff
@@ -115,7 +115,8 @@ def models_with_computedfields(self):
         for model in self.models:
             fields = set()
             for field in model._meta.fields:
-                if field in self.computedfields:
+                # We're excluding fields that are defined on non-abstract parent models
+                if field.model == model and field in self.computedfields:
```

Comment:
Ok thats weird - I actually started the resolver rewrite with your added clause here only, but got lots of test errors because of abstract models, therefore I reworked it that way.

Does this still work with abstract model base classes?

Reply:
Yes it does, well I mean the Abstract Model test cases still pass, and from what I've seen, the `model` attribute for Abstract model fields points to the Concrete model, so they're not being excluded by this test.

In fact I realized while implementing this PR that one of the "Abstract" model base classes used in tests wasn't abstract at all: I had forgotten to set the Meta abstract flag !

This is probably why when you added this check some of the Abstract model tests started to fail

Anyway, I added the abstract flag to this class and the tests still passed.

You can see that change in this PR

Reply:
> In fact I realized while implementing this PR that one of the "Abstract" model base classes used in tests wasn't abstract at all: I had forgotten to set the Meta abstract flag !

Oh well, might explain that, yeah. I mostly use the test cases for semi high level regression testing, better test cases would intercept the depends rules to field deps mapping and graph level already. Imho most test cases simply work, because they rely on full `save()` calls, which updates everything for a particular instance, while the `update_fields` stuff was added later. It is def. missing test cases. This also led to a shift in the graph shaping, where I had to introduce explicit auto-expansion on related fields to correctly propagate fields in question.